src/iop/ashift.c

```diff
@@ -1285,18 +1285,17 @@ void modify_roi_in(struct dt_iop_module_t *self,
 
   const struct dt_interpolation *interpolation =
     dt_interpolation_new(DT_INTERPOLATION_USERPREF_WARP);
-  roi_in->x = fmaxf(0.0f, xm - interpolation->width);
-  roi_in->y = fmaxf(0.0f, ym - interpolation->width);
-  roi_in->width = fminf(ceilf(orig_w) - roi_in->x,
-                        xM - roi_in->x + 1 + interpolation->width);
-  roi_in->height = fminf(ceilf(orig_h) - roi_in->y,
-                         yM - roi_in->y + 1 + interpolation->width);
+
+  roi_in->x       = xm - interpolation->width;
+  roi_in->y       = ym - interpolation->width;
+  roi_in->width   = xM + interpolation->width - xm + 1;
+  roi_in->height  = yM + interpolation->width - ym + 1;
```

Comment:
The old code, expanded out, does `+ 2 * interpolation->width`.  So this is not computing exactly the same size even without the clamping, and my immediate reaction is that we need the doubling since the interpolation will access beyond the edge on both sides.

Reply:
Your immediate reaction is absolutely correct!

Reply:
fixed.